course6/toyapp-vivike/displaylib/objects.h

```diff
@@ -14,44 +15,79 @@ struct Color
 	void redder(int amount);
 };
 
-struct Vector2D
-{
-	int x;
-	int y;
-
-	void move(int xmove, int ymove);
-	
-	Vector2D operator +(Vector2D b) { return { x + b.x, y + b.y}; }
-	Vector2D operator +(int b) { return { x + b, y + b}; }	
-
-	Vector2D operator *(int b) { return { x * b, y * b}; }
-
-	Vector2D operator -(Vector2D r) { return { x - r.x, y - r.y}; }
-	Vector2D operator -(int r) { return { x - r, y - r}; }
 
-	Vector2D operator /(int r) { return { x / r, y / r}; }
-};
-
-struct Vector3D
+template<typename T, int length>
+class Vector
 {
-	int x;
-	int y;
-	int z;
+public:
+	std::array<T, length> vec;
+
+	//Vector(std::array<T, length> newvec): vec(newvec) {};
```

Comment:
This would work, you'd have to use it like that:
```
Vector b({1, 2, 3});
```

Alternatively, you can make the constructor a variadic template. Look at the code of other participants.